nvgpu/list_gpus.py

```diff
@@ -17,7 +17,11 @@ def device_status(device_index):
     if six.PY3:
         device_name = device_name.decode('UTF-8')
     nv_procs = nv.nvmlDeviceGetComputeRunningProcesses(handle)
-    utilization = nv.nvmlDeviceGetUtilizationRates(handle).gpu
+    try:
+        utilization = nv.nvmlDeviceGetUtilizationRates(handle).gpu
+    except nv.NVMLError:
+        utilization = 'N/A'
```

Comment:
Possibly just fall back to `None` to represent a missing value. Then convert to `"N/A"` only for the purpose of printing.

Below:
```
return {
        'type': device_name,
        'is_available': len(pids) == 0,
        'pids': ','.join([str(pid) for pid in pids]),
        'users': ','.join(users),
        'running_since': arrow.get(min(dates)).humanize() if len(dates) > 0 else None,
        'utilization': utilization if utilization is not None else "N/A",
        'clock_mhz': clock_mhz,
        'temperature': temperature,
        'cmd': cmd,
        }
    }
```

Reply:
Then I would propose to just do:

```
return {
        'type': device_name,
        'is_available': len(pids) == 0,
        'pids': ','.join([str(pid) for pid in pids]),
        'users': ','.join(users),
        'running_since': arrow.get(min(dates)).humanize() if len(dates) > 0 else None,
        'utilization': utilization,
        'clock_mhz': clock_mhz,
        'temperature': temperature,
        'cmd': cmd,
        }
}
```

And perform the formatting  in `format_table` like so:

`df['util.'] = df['utilization'].apply(lambda u: '%03s %%' % u if u is not None else 'N/A')`

Reply:
Yeah, that sound better.